moveit_simple/src/moveit_simple.cpp

```diff
@@ -775,6 +775,44 @@ bool Robot::getIK(const Eigen::Affine3d pose, std::vector<double> & joint_point,
 
 
 
+bool Robot::isNearSingular(const std::vector<double> & joint_point) const
+{
+  std::lock_guard<std::recursive_mutex> guard(m_);
+  double threshold = 1e-10; // How to determine this threshold?
```

Comment:
I am not able to find any reference on this which talks about this threshold. What is the expected angle difference from the singular configuration that you think we should target to trigger isNearSingular? That can help in deciding a safe threshold.